src/helpers/overview-table/survey-data-type.ts

```diff
@@ -0,0 +1,50 @@
+import { DailyDataType } from "../daily-data-types";
+import { DailyDataProvider, DailyDataQueryResult } from "../query-daily-data";
+import { SurveyAnswersQuery } from "@careevolution/mydatahelps-js";
+import getDayKey from "../get-day-key";
+import { add, isSameDay, parseISO } from "date-fns";
+import queryAllSurveyAnswers from "../query-all-survey-answers";
+
+export interface SurveyDataType {
+    surveyName: string;
+    stepIdentifier: string;
+    resultIdentifier?: string;
+}
+
+export function isSurveyDataType(dataType: DailyDataType | SurveyDataType): dataType is SurveyDataType {
+    return dataType && (dataType as any).surveyName !== undefined;
+}
+
+export function getSurveyDataProvider(dataType: SurveyDataType): DailyDataProvider {
+    return async (startDate: Date, endDate: Date) => {
+        const query: SurveyAnswersQuery = {
+            surveyName: dataType.surveyName,
+            stepIdentifier: dataType.stepIdentifier,
+            after: add(startDate, { days: -1 }).toISOString(),
```

Comment:
Why are days being added to?

Reply:
I usually do that just to ensure that we get all the data we want back from the query, without having to worry about any cross-platform (client/server/db) timezone offset nuances around the day boundaries.  It probably isn't necessary, but, at least in this case, the extra days only add 1 or 2 more survey answers to the returned result, so it is relatively harmless for the assurance it provides.

I think it is a pretty common pattern in use by the majority of our other daily data providers, too, possibly for a similar reason.